### ISSUE TITLE
Corner case of integer variables with boundaries

### ISSUE THREAD

**Issue by nikohansen** [CMA-ES/pycma#260]
Related to #259.

For an upper variable bound at 1.5, an infeasible value of 1.6 is repaired to 1.5 which is then rounded to 2.0 (which is not an issue with `BoundTransform.repair`). If 1 is the optimum, this means that the optimum is not evaluated anymore (for unjustifiable reasons) unless the mean gets feasible again.

~This is also a potential issue on the `bbob-mixint` test suite (with `BoundPenalty` handling).~

A possible fix is to amend the bounds for integer variables (phenotype indices) by amending the bounds option.

**Comment by nikohansen** [comment]
In the `integer_centering` module the corner case is addressed like
```python
                idx = np.mod(bounds, 1) == 0.5
                bounds[idx] += (1 if which == 'lower' else
                    -1) * self.params['bounds_offset'] * np.maximum(1, np.abs(bounds[idx]))
```

**Comment by nikohansen** [comment]
Fixed with commit 8b117379.
